### PR TITLE
[Native][AmplifyAnalytic][AmplifyStorage] doc updates 

### PR DESCRIPTION
**Pull request by DongQuanRui**
*Description of changes:*
In this PR, I update analytics docs, mostly restructure the docs and change the code because of Amplify.Auth. And also work on example code part of Amplify.Storage.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/analytics/fragments/android/getting-started/20_installLib.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    dependencies {
+        classpath 'com.android.tools.build:gradle:3.5.0'
+    }
+}
+```
+
+Next add the following dependencies to your app `build.gradle`:
+
+```groovy
+implementation 'com.amplifyframework:core:0.10.0'
```
Comment (jpignata) [inline docs/lib/analytics/fragments/android/getting-started/20_installLib.md:19]: 1.0.0

File: docs/lib/analytics/fragments/android/getting-started/20_installLib.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+Open your project `build.gradle` and add the following:
+
+* `mavenCentral()` as a repository
```
Comment (jpignata) [inline docs/lib/analytics/fragments/android/getting-started/20_installLib.md:3]: This won't be necessary if they follow the Project Setup guide

File: docs/lib/analytics/fragments/android/getting-started/20_installLib.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+Open your project `build.gradle` and add the following:
```
Comment (jpignata) [inline docs/lib/analytics/fragments/android/getting-started/20_installLib.md:1]: I'd remove this.

File: docs/lib/analytics/fragments/android/getting-started/30_initAnalytics.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+Add the following imports to the top of your `MainActivity.java` file:
+
+```java
+import com.amplifyframework.analytics.pinpoint.AmazonPinpointAnalyticsPlugin;
```
Comment (jpignata) [inline docs/lib/analytics/fragments/android/getting-started/30_initAnalytics.md:4]: We can rely on their IDE to add these.

File: docs/lib/analytics/fragments/android/getting-started/30_initAnalytics.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        AmazonCognitoAuthPlugin authPlugin = new AmazonCognitoAuthPlugin((Application) context)
+        AmazonPinpointAnalyticsPlugin analyticsPlugin = new AmazonPinpointAnalyticsPlugin((Application) context);
+        Amplify.addPlugin(authPlugin)
+        Amplify.addPlugin(analyticsPlugin);
+        Amplify.configure(context);
+    } catch (Exception e) {
+        Log.e("GetStarted", "Error initializing", e);
+    }
+
+    Analytics.recordEvent("GetStarted");
```
Comment (jpignata) [inline docs/lib/analytics/fragments/android/getting-started/30_initAnalytics.md:23]: Let's make this event named something else to make it obvious

File: docs/lib/analytics/fragments/autotrack.md [hunk 6]
```diff
@@ -3,9 +3,8 @@ The Amplify analytics plugin records when an application opens and closes. This
...
 
 ```bash
-amplify console
+amplify console analytics
 ```
 
-1. Navigate to the *Analytics* tab, and then choose *View in Pinpoint*.
-2. On the Pinpoint Console under Analytics, choose *Events*. 
-3. Enable filters, you can select `Session Start` and `Session Stop` events to filter on session events.
+1. On the Pinpoint Console under Analytics, choose *Events*.
```
Comment (jpignata) [inline docs/lib/analytics/fragments/autotrack.md:9]: Change all of these to 1. for maintainability.

Comment (jpignata) [inline docs/lib/analytics/fragments/autotrack.md:9]: Can we embolden **Events**?

File: docs/lib/analytics/fragments/ios/getting-started/30_initAnalytics.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import Amplify
+import AmplifyPlugins
+```
+
+**Add the following code** to your AppDelegate's `application:didFinishLaunchingWithOptions` method:
+```swift
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+
+    do {
+        try Amplify.add(plygin: AWSCognitoAuthPlugin())
```
Comment (jpignata) [inline docs/lib/analytics/fragments/ios/getting-started/30_initAnalytics.md:15]: Typo.

File: docs/lib/analytics/fragments/ios/getting-started/30_initAnalytics.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+        try Amplify.add(plugin: AWSPinpointPlugin())
+        try Amplify.configure()
+        print("Amplify configured with auth and analytics plugin")
```
Comment (jpignata) [inline docs/lib/analytics/fragments/ios/getting-started/30_initAnalytics.md:18]: plugins

File: docs/lib/analytics/fragments/ios/getting-started/30_initAnalytics.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    } catch {
+        print("Failed to initialize Amplify with \(error)")
+    }
+
+    return true
+}
+```
+Upon building and running this application you should see the following in your console window:
+
+```bash
```
Comment (jpignata) [inline docs/lib/analytics/fragments/ios/getting-started/30_initAnalytics.md:28]: This isn't bash. Use `console`.

File: docs/lib/analytics/fragments/native_common/getting-started/common.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+Run the following command in your project's root folder. The CLI will prompt configuration options for the Analytics category such as Amazon Pinpoint resource name and analytics event settings.
+
+> The Analytics category utilizes the Authentication category behind the scenes to authorize your app to send analytics events.
+
+```bash
+amplify add analytics
+```
+
+```console
+? Provide your pinpoint resource name: `yourPinpointResourceName`
```
Comment (jpignata) [inline docs/lib/analytics/fragments/native_common/getting-started/common.md:19]: Let's put these on the next line.

**Comment by DongQuanRui** [comment]
Hi, John P. So the issue is I don't know Java, That's why I requested Roshan to review it. And hope he can help to improve Android docs. I only did a set up for Android. For iOS, I will go and fix it right now

**Comment by jpignata** [comment]
> Hi, John P. So the issue is I don't know Java, That's why I requested Roshan to review it. And hope he can help to improve Android docs. I only did a set up for Android. For iOS, I will go and fix it right now

That's fine. I can help out with the Java bits. Let's just do some clean-up, slim this down, and polish it.

File: docs/lib/analytics/fragments/android/getting-started/20_installLib.md (hunk 3 above)
Comment (DongQuanRui) [reply docs/lib/analytics/fragments/android/getting-started/20_installLib.md:1]: > > Hi, John P. So the issue is I don't know Java, That's why I requested Roshan to review it. And hope he can help to improve Android docs. I only did a set up for Android. For iOS, I will go and fix it right now
> 
> That's fine. I can help out with the Java bits. Let's just do some clean-up, slim this down, and polish it.

Cool, let me split this PR first

File: docs/lib/analytics/fragments/native_common/getting-started/common.md (hunk 10 above)
Comment (DongQuanRui) [reply docs/lib/analytics/fragments/native_common/getting-started/common.md:19]: Do you mean by putting an empty line between "```console" and "? Provide your *"

**Pull request closed without merging**